autorally_core/src/StateEstimator/StateEstimator.cpp

```diff
@@ -284,42 +284,113 @@ namespace autorally_core
     gyro = Vector3(gx, gy, gz);
   }
 
-  void Imu_Gps::GpsHelper()
+  void StateEstimator::GpsHelper()
   {
+    double prevTime = 0;
+    double curTime = 0;
+    double timeWithoutGPS = 0;
+    unsigned char status = autorally_msgs::stateEstimatorStatus::OK;
+    ros::Rate loop_rate(m_frequency * 1.25);
+    int gpsErrorCount = 0;
+
     // Kick off the thread, and wait for our GPS measurements to come streaming in
     while (ros::ok())
     {
-      sensor_msgs::NavSatFixConstPtr fix = m_gpsOptQ.popBlocking();
-      ++m_gpsCounter;
-      if (m_gpsCounter > m_gpsSkip) m_gpsCounter = 0;
-      else continue;
+      sensor_msgs::NavSatFixConstPtr fix;
+      bool usingGPS = false;
+      bool usingOdom = false;
+      bool usingIMU = false;
+
+
+      // remove any old GPS measurements
```

Comment:
Is it possible this will cause us to throw out gps measurements that have a place in the graph?